docs/FeatureManagement/Clients/Microsoft.FeatureManagement.AspNetCore.md

```diff
@@ -0,0 +1,367 @@
+
+When an MVC controller or action is blocked because none of the features it specifies are enabled, then the registered `IDisabledFeatureHandler` is triggered. The default `IDisabledFeatureHandler` returns a 404 status code to the client with no response body. This behavior can be overridden when registering the feature management services.
+
+```
+public interface IDisabledFeatureHandler
+{
+    Task HandleDisabledFeature(string featureName, ActionExecutingContext context);
+}
+```
+
+### View
+
+In MVC views `<feature>` tags can be used to conditionally render content based on whether a feature is enabled or not.
+
+```
+<feature name="FeatureX">
+  <p>This can only be seen if 'FeatureX' is enabled.</p>
+</feature>
+```
+
+### MVC Filters
+
+MVC filters can be set up to conditionally execute based on the state of a feature. This is done by registering MVC filters in a feature aware manner.
+
+```
+services.AddMvc(o => 
+{
+    o.Filters.AddForFeature<SomeMvcFilter>(nameof(MyFeatureFlags.FeatureV));
+});
+```
+
+The code above adds an MVC filter named `SomeMvcFilter`. This filter is only triggered within the MVC pipeline if the feature it specifies, "FeatureV", is enabled.
+
+### Routing
+Certain routes may expose application capabilites that are gated by features. These routes can be dynamically registered and exposed based on whether a feature is enabled.
+
+```
+app.UseMvc(routes =>
+{
+    routes.MapRouteForFeature(nameof(MyFeatureFlags.Beta), "betaDefault" /* route name */, "{controller=Beta}/{action=Index}/{id?}"
+});
+```
+
+The route above exposes beta functionality of some application that is gated behind the "Beta" feature flag. The state of this feature is evaluated on every request, which means routes can be exposed to a subset of requests and or users. Furthermore, routes can dynamically be added or removed based on the toggling of feature states.
+
+### Application building
+
+The feature management library can be used to add application branches and middleware that execute conditionally based on feature state.
+
+```
+app.UseMiddlewareForFeature<ThirdPartyMiddleware>(nameof(MyFeatureFlags.FeatureU));
+```
+
+With the above call, the application adds a middleware component that only appears in the request pipeline if the feature "FeatureU" is enabled. If the feature is enabled/disabled during runtime, the middleware pipeline can be changed dynamically.
+
+This builds off the more generic capability to branch the entire application based on a feature.
+
+```
+app.UseForFeature(featureName, appBuilder => 
+{
+    appBuilder.UseMiddleware<T>();
+});
+```
+
+### Disabled Action Handling
+
+When an MVC action requires to be enabled and the feature is not enabled, a registered `IDisabledFeatureHandler` will be invoked. By default, a minimalistic handler is registered which returns HTTP 404. This can be overridden using the `IFeatureManagmentBuilder` when registering feature flags.
```

Comment:
We talked about this in the end of "Controllers and Actions" section too. Can we combine them?